corehq/apps/hqwebapp/forms.py

```diff
@@ -21,6 +24,23 @@ def clean_username(self):
         username = self.cleaned_data['username'].lower()
         return username
 
+    def clean(self):
+        lockout_message = mark_safe(_('Sorry - you have attempted to login with an incorrect password too many times. Please <a href="/accounts/password_reset_email/">click here</a> to reset your password.'))
+        username = self.cleaned_data['username']
+        try:
+            cleaned_data = super(EmailAuthenticationForm, self).clean()
+        except ValidationError:
+            user = CouchUser.get_by_username(username)
+            if user and user.is_web_user() and user.is_locked_out():
+                raise ValidationError(lockout_message)
+            else:
+                raise
+        user = CouchUser.get_by_username(username)
+        if user and user.is_web_user() and user.is_locked_out():
+            raise ValidationError(lockout_message)
```

Comment:
Looks like lines 33-35 and 38-50 are same. Can one of this be placed before `try-except` block to avoid repetition and also the need for `try-except` block?